### PR TITLE
Tapiro core tests

### PR DESCRIPTION
**Pull request by gabro**
Based on #210, only the last commit is relevant.

## TODO
- add more tests if we like the approach
- ~add a NOTICE for the code I've copy-pasted from Metals~
- logs are not displayed when running the sbt plugin

### CODE REVIEW

File: tapiro/core/src/test/scala/io/buildo/tapiro/FileLayout.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package io.buildo.tapiro
```
Comment (gabro) [inline tapiro/core/src/test/scala/io/buildo/tapiro/FileLayout.scala]: this file is copy-pasted from `Metals` almost verbatim

File: tapiro/core/src/test/scala/io/buildo/tapiro/TapiroSuite.scala [hunk 2]
```diff
@@ -0,0 +1,135 @@
+package io.buildo.tapiro
+
+import java.nio.file.Files
+
+class TapiroSuite extends munit.FunSuite {
+
+  check(
```
Comment (gabro) [inline tapiro/core/src/test/scala/io/buildo/tapiro/TapiroSuite.scala]: This is a test, the idea is that you pass in the controller code and the test runs tapiro and checks the expected code against the generated one.

File: tapiro/core/src/test/scala/io/buildo/tapiro/TapiroSuite.scala [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      |  case object NotFound extends SchoolReadError
+      |}
+      |
+      |trait SchoolController[F[_], T] {
+      |  @query
+      |  def read(id: Long): F[Either[SchoolReadError, School]]
+      |}
+      |""".stripMargin,
+    """
+      |/src/main/scala/schools/endpoints/SchoolControllerTapirEndpoints.scala
```
Comment (gabro) [inline tapiro/core/src/test/scala/io/buildo/tapiro/TapiroSuite.scala]: line starting with a single `/` declare files that are written to disk (in a temp dir)

File: tapiro/core/src/main/scala/io/buildo/tapiro/Util.scala [hunk 4]
```diff
@@ -36,9 +33,11 @@ object TapiroRouteError {
 
 case class TapiroRoute(route: Route, error: TapiroRouteError)
 
-class Util(logger: ManagedLogger) {
+class Util() {
   import Formatter.format
 
+  val logger = LogManager.getLogger("io.buildo.tapiro")
```
Comment (gabro) [inline tapiro/core/src/main/scala/io/buildo/tapiro/Util.scala]: this does not show up when running the sbt plugin, I'll open a separate issue for it

**Comment by tpetrucciani** [comment]
I've tried to rebase my PR #212 on this and add more tests (not pushed yet).

I think it works quite nicely so I'd be in favour of merging this as is (bar the conflict to fix) and keep the "add more tests" part for #212 since tests on POST endpoints would have to be updated anyway. @gabro what do you think?

@calippo did you get a chance to look at this?

**Comment by gabro** [comment]
I'm 👍 for merging it as is. This needs a rebase, but it should be easy.

**Comment by calippo** [comment]
I had a quick look, lgtm

**Comment by gabro** [comment]
Rebased, we can merge as soon as the CI passes

**Pull request merged**